src/libbson/src/bson/bson.c

```diff
@@ -3657,7 +3657,7 @@ struct _bson_array_builder_t {
 bson_array_builder_t *
 bson_array_builder_new (void)
 {
-   bson_array_builder_t *bab = bson_malloc0 (sizeof (bson_array_builder_t));
+   bson_array_builder_t *bab = bson_aligned_alloc0 (128, sizeof (bson_array_builder_t));
```

Comment:
```suggestion
   bson_array_builder_t *bab = BSON_ALIGNED_ALLOC0 (bson_array_builder_t);
```

Suggest using the macro form to determine the necessary alignment. The alignment may not always be 128. libbson may be configured with the CMake option `ENABLE_EXTRA_ALIGNMENT=OFF` to disable the over-alignment of `bson_t`.